src/modules/rating/rating.ts

```diff
@@ -0,0 +1,22 @@
+import { ObjectId } from 'mongodb';
+import { useCollection } from 'db';
+import type { RatingForm } from 'prytaneum-typings';
+
+import createHttpError from 'http-errors';
+
+const addRating = async (
+    rating: RatingForm,
+    townhallId: string
+): Promise<void> => {
+    const filter = { _id: new ObjectId(townhallId) };
+    const update = { $push: { ratings: rating } };
+    const options = { upsert: true, returnOriginal: true };
+    const doc = await useCollection('Ratings', (Ratings) => {
+        return Ratings.findOneAndUpdate(filter, update, options);
+    });
+    if (!doc) throw createHttpError(404, 'Townhall doc not found');
+};
```

Comment:
Lets do a few things:
1. Right now the relationship between the `Ratings` collection and the `Townhalls` collection is 1-to-1.  Lets make this many-to-one since there's technically no upper bound to the amount of ratings a particular townhall might have.  It is bounded by the number of attendees, but we don't strictly limit the maximum amount of attendees.  If this relationship were few-to-one, think something like the number of moderators in a townhall, where the number of moderators would never exceed ~100, then we would just embed the field right into the townhall collection object.  For reference, the memory limit of one document in a mongodb collection is 16mb.
2. The rating `_id` should probably be different from the `Townhalls` `_id`. The Ratings collection object should have a `meta` field with the `townhallId` field since that is what we have been doing for the other collections.

Reply:
I fixed the _id/townhallId issue, but I'm a little confused on how you want me to modify it, do you want each rating to be a new doc and do like a query on the metadata when getting the ratings for a particular townhall?

Reply:
That's a really good question.  I hadn't thought of that.   I think we'll need to make a custom meta field then for the rating.  It should be fairly simple: 
```ts
const RatingMetaAt = Pick<Meta, 'createdAt' | 'updatedAt'>;
const RatingMetaBy = Partial<Pick<Meta, 'createdBy' | 'updatedBy'>>
const RatingMeta< T extends string | ObjectId = string> =  RatingMetaAt & RatingMetaBy & { townhallId: T }; 
```
Let me know if you'd like me to explain this ts further.